src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt

```diff
@@ -2062,51 +2074,74 @@ namespace NServiceBus.Pipeline.Contexts
+        public void AddStatic(System.Type messageType, NServiceBus.EndpointInstanceName destination) { }
+        public void AddStatic(System.Type messageType, string destinationAddress) { }
     }
-    public class DynamicRoutingProvider
+    public class DistributionPolicy
     {
-        public DynamicRoutingProvider() { }
-        public NServiceBus.Routing.IProvideDynamicRouting DynamicRouting { get; set; }
-        public string GetRouteAddress(string endpoint) { }
+        public void SetDistributionStrategy(NServiceBus.Routing.DistributionStrategy distributionStrategy, System.Func<System.Type, bool> typeMatchingRule) { }
     }
-    public interface IProvideDynamicRouting
+    public abstract class DistributionStrategy
     {
-        bool TryGetRouteAddress(string logicalEndpoint, out string address);
+        protected DistributionStrategy() { }
+        public abstract System.Collections.Generic.IEnumerable<NServiceBus.EndpointInstanceName> SelectDestination(System.Collections.Generic.IEnumerable<NServiceBus.EndpointInstanceName> allInstances);
     }
-    public class static RoutingContextExtensions
+    public class IndirectRoutingSlip : NServiceBus.Routing.RoutingSlip
+    {
+        public IndirectRoutingSlip(System.Type messageType) { }
+        public System.Type MessageType { get; }
+        public override void Serialize(System.Collections.Generic.Dictionary<string, string> options) { }
+    }
+    public class KnownEndpoints
     {
-        public static NServiceBus.Routing.RoutingStrategy GetRoutingStrategy(this NServiceBus.TransportDispatch.DispatchContext context) { }
+        public KnownEndpoints() { }
+        public void AddDynamic(System.Func<NServiceBus.EndpointName, System.Collections.Generic.IEnumerable<NServiceBus.EndpointInstanceName>> dynamicRule) { }
+        public void AddStatic(NServiceBus.EndpointName endpoint, params NServiceBus.EndpointInstanceName[] instances) { }
+        public void AddStaticUsingTransportDiscriminators(NServiceBus.EndpointName endpoint, params string[] transportDiscriminators) { }
     }
-    public class RoutingExtensions<T> : NServiceBus.Configuration.AdvanceExtensibility.ExposeSettings
-        where T : NServiceBus.Routing.DynamicRoutingDefinition
+    public class static RoutingContextExtensions
     {
-        public RoutingExtensions(NServiceBus.Settings.SettingsHolder settings) { }
+        public static System.Collections.Generic.IEnumerable<NServiceBus.Routing.RoutingSlip> GetRoutingSlips(this NServiceBus.TransportDispatch.DispatchContext context) { }
+        public static void SetRoutingSlips(this NServiceBus.Pipeline.BehaviorContext context, System.Collections.Generic.IEnumerable<NServiceBus.Routing.RoutingSlip> routingSlips) { }
+        public static void SetRoutingSlips(this NServiceBus.Pipeline.BehaviorContext context, params NServiceBus.Routing.RoutingSlip[] routingSlips) { }
     }
-    public abstract class RoutingStrategy
+    public abstract class RoutingSlip
```

Comment:
RoutingSlip in my opinion is a misleading name for what this does. RoutingSlip and i immediately associate it with the itinerary pattern.